### PR TITLE
MULE-17110: NTLM authentication with dynamic credentials is successful despite credentials changes

### PR DESCRIPTION
**Pull request by plbalbi**

### CODE REVIEW

File: src/test/java/org/mule/service/http/impl/NtlmMockResponseGenerator.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    // Client does not know the target service is requires authentication
+    if (authorization == null) {
+      responseBuilder.statusCode(UNAUTHORIZED.getStatusCode());
+      responseBuilder.addHeader(WWW_AUTHENTICATE, NTLM);
+      // Client knows that NTLM Authentication is required
+    } else if (authorization.equals(this.firstMessageExpectedHeader)) {
+      responseBuilder.statusCode(UNAUTHORIZED.getStatusCode());
+      responseBuilder.addHeader(WWW_AUTHENTICATE, this.challenge);
+      this.currentState = State.CHALLENGE_SENT;
+      // First message with challenge sent, verifying outcome of dace
```
Comment (afelisatti) [inline src/test/java/org/mule/service/http/impl/NtlmMockResponseGenerator.java]: dance

File: src/test/java/org/mule/service/http/impl/NtlmMockResponseGenerator.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+  private final String challenge;
+  private final String secondMessageExpectedHeader;
+
+  private final String username;
+  private final String password;
+  private final String domain;
+
+  private State currentState;
+
+  private final static String DEFAULT_CREDENTIALS_FILE = "ntlm-credentials.conf";
```
Comment (afelisatti) [inline src/test/java/org/mule/service/http/impl/NtlmMockResponseGenerator.java]: but why? do you change these anywhere?

Comment (plbalbi) [reply src/test/java/org/mule/service/http/impl/NtlmMockResponseGenerator.java]: They are reused between two test suites, and in it seemed like a good idea to reuse them if some others NTLM tests are added.

File: src/main/java/org/mule/service/http/impl/service/client/GrizzlyHttpClient.java [hunk 3]
```diff
@@ -397,6 +399,17 @@ protected Request createGrizzlyRequest(HttpRequest request, HttpRequestOptions o
...
+
+          // If Authentication has forceConnectionClose method, set it in the authentication Realm
+          try {
+            Method forceConnectionCloseMethod = authentication.getClass().getMethod(FORCE_CONNECTION_CLOSE);
+            // To prevent cross-package access exceptions
+            forceConnectionCloseMethod.setAccessible(true);
+            realmBuilder.setForceConnectionClose((boolean) forceConnectionCloseMethod.invoke(authentication));
+          } catch (Exception e) {
+            logger
+                .warn("The authentication object used configuring an NTLM authentication does not implement 'forceConnectionClose'.");
```
Comment (afelisatti) [inline src/main/java/org/mule/service/http/impl/service/client/GrizzlyHttpClient.java]: debug?

Comment (plbalbi) [reply src/main/java/org/mule/service/http/impl/service/client/GrizzlyHttpClient.java]: Yep. Less overhead.

**Pull request closed without merging**